### DISCUSSION TITLE
Help me: the way to run(import) JSX on nodeJS server

### DISCUSSION THREAD

**Discussion by botirk** [babel/babel#13684] [Q&A] [answered]
`npx babel-node --presets @babel/preset-react server/start.js`
is my code to start server
```
"dependencies": {
    "@babel/core": "^7.15.0",
    "@babel/node": "^7.14.9",
    "@babel/preset-env": "^7.15.0",
    "@babel/preset-react": "^7.14.5",
    "fastify": "^3.20.2",
    "fastify-static": "^4.2.3",
    "is-mobile": "^3.0.0",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
```
are my dependencies

`import HelloWorld from './components/HelloWorld.jsx';` 
is my code to import jsx code
`Unknown file extension ".jsx"` 
is my error

**Please help me with the way to launch nodeJS server which imports jsx code and renders it using** 
`ReactDOMServer.renderToString` 
**function?**

**Comment by fedeci** [comment]
Have you tried omitting the extension in the `import`?

**Reply by botirk** [reply]
`Cannot find module '/home/ivan/code/mobileWebSite/server/components/HelloWorld'`
Error appears

**Reply by botirk** [reply]
It seems babel-node ignores "@babel/preset-react" :(

**Reply by fedeci** [reply]
Is the repo open source?

**Reply by botirk** [reply]
I have made it open source: https://github.com/botirk/mobileWebSite/

**Comment by nicolo-ribaudo** [comment, marked as answer]
What is the stack trace of the `Unknown file extension` error?

**Reply by botirk** [reply]
node:internal/process/esm_loader:74
    internalBinding('errors').triggerUncaughtException(
                              ^

TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".jsx" for /home/ivan/code/mobileWebSite/server/components/HelloWorld.jsx
    at new NodeError (node:internal/errors:329:5)
    at Loader.defaultGetFormat [as _getFormat] (node:internal/modules/esm/get_format:71:15)
    at Loader.getFormat (node:internal/modules/esm/loader:104:42)
    at Loader.getModuleJob (node:internal/modules/esm/loader:242:31)
    at async ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:59:21)
    at async Promise.all (index 6)
    at async link (node:internal/modules/esm/module_job:64:9) {
  code: 'ERR_UNKNOWN_FILE_EXTENSION'
}

**Reply by nicolo-ribaudo** [reply]
The problem is that you are using native ECMAScript modules, which are not supported by `babel-node`. See https://github.com/babel/babel/issues/11934 for more info.

**Reply by botirk** [reply]
Ok! I will redo **import** into **require**
